ivy/functional/backends/torch/general.py

```diff
@@ -187,31 +187,82 @@ def gather(
     out: Optional[torch.Tensor] = None,
 ) -> torch.Tensor:
     axis %= len(params.shape)
+    axis = abs(len(params.shape) + axis) if axis < 0 else axis
```

Comment:
Just a small point here, wouldn't `axis % len(params.shape)` always be positive? I suppose we wouldn't need the `if` check in that case?

Reply:
I'll check that but I think there was some issue with 'torch.expand()' in the batching part or in slicing- or some other element, I don't remember right now. I only remember that I added that because of the batching in general.